### PR TITLE
update error logging

### PR DESCRIPTION
**Pull request by ratik21**

### CODE REVIEW

File: packages/algob/src/internal/core/errors.ts [hunk 1]
```diff
@@ -190,16 +190,16 @@ function _applyErrorMessageTemplate (
...
 
-  if (e?.statusCode !== undefined) {
-    if (e.statusCode >= 400 && e.statusCode < 500) {
+  if (e.response?.statusCode !== undefined) {
+    if (e.response?.statusCode >= 400 && e.response?.statusCode < 500) {
       return new BuilderError(ERRORS.ALGORAND.BAD_REQUEST, {
-        status: e.statusCode,
-        message: e.body?.message ?? e.text,
+        status: e.response?.statusCode,
+        message: e.response?.body?.message ?? e.response?.text,
```
Comment (robert-zaremba) [inline packages/algob/src/internal/core/errors.ts:197]: please add a check for `e.error.text` (or `e.response.error.text` - not sure which one is correct) as a third option if the 2 previous are not defined.

**Review by robert-zaremba — APPROVED** [review]
left one comment.

File: packages/algob/src/internal/core/errors.ts (hunk 1 above)
Comment (ratik21) [reply packages/algob/src/internal/core/errors.ts:197]: Sorry, i don't follow. I think the `response` part was added after the latest sdk update. Line 197 is checking both possibilites (`body.message` and `text`). 
BDW, this block will execute only if the error is `Bad Request`.

Comment (robert-zaremba) [reply packages/algob/src/internal/core/errors.ts:197]: I was thinking about:
```
e.response?.body?.message ?? e.response?.text ?? e.response?error
```
Although I don't remember if the last part is `e.response?error` or `e.error`

Comment (ratik21) [reply packages/algob/src/internal/core/errors.ts:197]: ok. It's `e.response?error`

**Pull request merged**